### PR TITLE
COTR-65 further GUI improvements

### PR DESCRIPTION
**Pull request by Arnelys**
Added better health bar styling.

Added golden coin popup animation when change currency method is called.

### CODE REVIEW

File: Scenes/UI/HUD/GUI.cs [hunk 1]
```diff
@@ -105,15 +112,24 @@ private void ChangeMaxHealth(float newHealthPoints)
...
     /// <summary>
-    /// Method to change the currency value
+    /// Method to change the currency value and play specific animations
     /// </summary>
     /// <param name="newCurrency">New currency value</param>
     private void ChangeCurrency(int oldCurrency, int newCurrency)
     {
+        goldenCoinSprite.Set("modulate", new Color(1,1,1,1));
+        goldenCoinSprite.Set("position", new Vector2(baseGoldCoinCoordinates.X, baseGoldCoinCoordinates.Y));
+
```
Comment (Elijuszek) [inline Scenes/UI/HUD/GUI.cs:122]: Add comment "Tweening gold text"

File: Scenes/UI/HUD/GUI.cs [hunk 2]
```diff
@@ -105,15 +112,24 @@ private void ChangeMaxHealth(float newHealthPoints)
...
         Tween goldTween = CreateTween();
         goldTween.SetTrans(Tween.TransitionType.Expo);
         goldTween.SetEase(Tween.EaseType.Out);
         goldTween.TweenMethod(new Callable(this, "ChangeGoldText"), oldCurrency, newCurrency, 0.6f);
+
```
Comment (Elijuszek) [inline Scenes/UI/HUD/GUI.cs:127]: Add comment "Tweening golden coin"

File: Scenes/UI/HUD/HUD.tscn [hunk 3]
```diff
@@ -1,21 +1,26 @@
-[gd_scene load_steps=7 format=3 uid="uid://xpy4rcjasw1t"]
+[gd_scene load_steps=8 format=3 uid="uid://xpy4rcjasw1t"]
 
 [ext_resource type="Script" path="res://Scenes/UI/HUD/GUI.cs" id="1_ujh52"]
+[ext_resource type="Texture2D" uid="uid://bkdy38u5jlvjk" path="res://Sprites/UI/Currency/GoldenCoin.png" id="2_fw1db"]
```
Comment (Elijuszek) [inline Scenes/UI/HUD/HUD.tscn:4]: `GoldenCoin.png` should be moved to `res://Sprites/Loot/Coins/GoldenCoin.png`

File: Scenes/UI/HUD/GUI.cs [hunk 4]
```diff
@@ -25,6 +26,8 @@ public partial class GUI : Control
     private Button[] itemArray;
     private int currentItemIndex;
 
+    private Vector2 baseGoldCoinCoordinates;
```
Comment (Elijuszek) [inline Scenes/UI/HUD/GUI.cs:29]: Comment why do you need this variable

**Review by Elijuszek — CHANGES_REQUESTED** [review]

File: Scenes/UI/HUD/GUI.cs (hunk 1 above)
Comment (Arnelys) [reply Scenes/UI/HUD/GUI.cs:122]: Done

File: Scenes/UI/HUD/GUI.cs (hunk 2 above)
Comment (Arnelys) [reply Scenes/UI/HUD/GUI.cs:127]: Done

File: Scenes/UI/HUD/HUD.tscn (hunk 3 above)
Comment (Arnelys) [reply Scenes/UI/HUD/HUD.tscn:4]: Done

File: Scenes/UI/HUD/GUI.cs (hunk 4 above)
Comment (Arnelys) [reply Scenes/UI/HUD/GUI.cs:29]: Done

**Review by Elijuszek — DISMISSED** [review]
Unresolved conflicts

**Review by Karolinskis — APPROVED** [review]
LGTM

**Pull request merged**